### DISCUSSION TITLE
Did Backlash Compensation Need Linear Encoder To be Feedback ?

### DISCUSSION THREAD

**Discussion by compoundMR** [grblHAL/core#608] [General]
im trying use a backlash compensation , and first , when i try use it by just moving G1F100 X-0.1 and then G1F100 X0 it seems the backlash compe work normally , but after i use a long gcode , my 0 point is change and not consistent do u have any idea what happen is it because of inertia movement ? or backlash compensation need linear encoder or ballscrew encoder to be feedback ?

**Comment by terjeio** [comment]
> backlash compensation need linear encoder or ballscrew encoder to be feedback ?

No. The backlash compensation is only added on direction changes and based on the configured distance.

>  my 0 point is change and not consistent do u have any idea what happen is it because of inertia movement ?

Is the long movement a single movement or comprised of many small that also includes direction changes? And are you sure the configured backlash distance is correct?

**Reply by compoundMR** [reply]
thanks in advance for replying ,


im sure i get 0.02 backlash on my machine and i only activate X axis, and here is my gcode i use for test


[backlash Test.txt](https://github.com/user-attachments/files/17360284/backlash.Test.txt)

**Reply by terjeio** [reply]
When I test with 250 step/mm and an odd setting just above that I get 5 steps out of sync (by counting steps). What is your step/mm setting and how much is zero off?

**Reply by compoundMR** [reply]
sorry for late reply , im setting it on 500 step/mm ( 0.002 accuracy ) , and get 3 miss step per section ( make a counter pulse on another controller and count the steps) , and get alternatively solved by increasing the step/mm into 1000 step/mm (0.001 accuracy )
